### ISSUE TITLE
resources don't allow integer values

### ISSUE THREAD

**Issue by safuya** [prometheus-operator/prometheus-operator#3352]
These values should be valid according to https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/, but they are not:

```
resources:
  requests:
    memory: 600Mi
    cpu: 1
  limits:
    memory: 600Mi
    cpu: 1
```

The error received when trying to deploy is:

```
Error from server (Invalid): error when creating "/tmp/tmp.1HCp1dN1G3/teamname/manifests/common-alertmanager-alertmanager.yaml": Alertmanager.monitoring.coreos.com "main" is invalid: []: Invalid value: map[string]interface ...redacted output...: validation failure list:
 spec.resources.limits.cpu in body must be of type string: "integer"
 spec.resources.requests.cpu in body must be of type string: "integer"
```

**Comment by brancz** [comment]
That's strange indeed. It'll need some more investigation. For now use `1000m` I guess, but I agree this should work.

**Comment by ingvarm-gr** [comment]
Alternatively, `cpu: "1"` (in YAML `1` is likely to be an integer, but `"1"` never is).

**Comment by slashpai** [comment]
I would like to work on this

**Comment by paulfantom** [comment]
@slashpai awesome! If you need help, feel free to ping us in #prometheus-operator-dev channel on kubernetes slack.

**Comment by slashpai** [comment]
I am trying to reproduce this error so I ran `kubectl apply -f bundle.yaml` in kind cluster and tried to create alertmanager resource using https://github.com/prometheus-operator/kube-prometheus/blob/main/manifests/alertmanager-alertmanager.yaml but I updated cpu requests and limit to use `1` and `4` respectively and I see I am able to give that as value for the resources. Am I missing any info here to recreate the same issue as reported?

```yaml
░▒▓    ~  k get alertmanager main -n monitoring -o yaml                                                                1|1 ✘  2.7.17   1.16.4   2.7.3   at kind-pai ⎈  at 16:34:04  ▓▒░
apiVersion: monitoring.coreos.com/v1
kind: Alertmanager
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"monitoring.coreos.com/v1","kind":"Alertmanager","metadata":{"annotations":{},"labels":{"alertmanager":"main","app.kubernetes.io/component":"alert-router","app.kubernetes.io/name":"alertmanager","app.kubernetes.io/part-of":"kube-prometheus","app.kubernetes.io/version":"0.21.0"},"name":"main","namespace":"monitoring"},"spec":{"image":"quay.io/prometheus/alertmanager:v0.21.0","nodeSelector":{"kubernetes.io/os":"linux"},"podMetadata":{"labels":{"app.kubernetes.io/component":"alert-router","app.kubernetes.io/name":"alertmanager","app.kubernetes.io/part-of":"kube-prometheus","app.kubernetes.io/version":"0.21.0"}},"replicas":3,"resources":{"limits":{"cpu":1,"memory":"100Mi"},"requests":{"cpu":4,"memory":"100Mi"}},"securityContext":{"fsGroup":2000,"runAsNonRoot":true,"runAsUser":1000},"serviceAccountName":"alertmanager-main","version":"0.21.0"}}
  creationTimestamp: "2021-06-11T11:03:36Z"
  generation: 1
  labels:
    alertmanager: main
    app.kubernetes.io/component: alert-router
    app.kubernetes.io/name: alertmanager
    app.kubernetes.io/part-of: kube-prometheus
    app.kubernetes.io/version: 0.21.0
  name: main
  namespace: monitoring
  resourceVersion: "363592"
  uid: 73fce0dc-91da-48e3-bd2c-200b29f5509c
spec:
  image: quay.io/prometheus/alertmanager:v0.21.0
  nodeSelector:
    kubernetes.io/os: linux
  podMetadata:
    labels:
      app.kubernetes.io/component: alert-router
      app.kubernetes.io/name: alertmanager
      app.kubernetes.io/part-of: kube-prometheus
      app.kubernetes.io/version: 0.21.0
  replicas: 3
  resources:
    limits:
      cpu: 1
      memory: 100Mi
    requests:
      cpu: 4
      memory: 100Mi
  securityContext:
    fsGroup: 2000
    runAsNonRoot: true
    runAsUser: 1000
  serviceAccountName: alertmanager-main
  version: 0.21.0
```

**Comment by dgrisonnet** [comment]
This issue was most likely fixed as a side effect of #3694.

**Comment by slashpai** [comment]
Thank you @dgrisonnet
I see it allows both integer and strong format now. 

@paulfantom We can close this one then?
